enterprise/internal/campaigns/store_changesets.go

```diff
@@ -346,13 +346,14 @@ SELECT changesets.id,
 	changesets.updated_at,
 	max(ce.updated_at) AS latest_event,
 	changesets.external_updated_at,
-	r.external_service_id
+	min(esr.external_service_id)
 FROM changesets
 LEFT JOIN changeset_events ce ON changesets.id = ce.changeset_id
 JOIN campaigns ON campaigns.changeset_ids ? changesets.id::TEXT
 JOIN repo r ON changesets.repo_id = r.id
+LEFT JOIN external_service_repos esr ON r.id = esr.repo_id
 WHERE %s
-GROUP BY changesets.id, r.id
+GROUP BY changesets.id, r.id %s
```

Comment:
This might be clearer with the pattern below to indicate what the bare `%s` is for:

```suggestion
GROUP BY changesets.id, r.id
%s --- having
```